addons/mrp/tests/test_warehouse_multistep_manufacturing.py

```diff
@@ -784,3 +784,33 @@ def test_component_and_byproduct_on_transfers(self):
             else:
                 self.assertEqual(len(picking.move_ids_without_package), 2, "Should have 2 moves for: Comp1 and Comp2")
                 self.assertEqual([move.product_id for move in picking.move_ids_without_package.sorted('product_id')], [comp1, comp2])
+
+    def test_sales_order_with_mto_manufacturing(self):
+        self.env.ref('stock.route_warehouse0_mto').active = True
+        warehouse = self.env.ref('stock.warehouse0')
+        with Form(warehouse) as w:
+            w.manufacture_steps = 'pbm_sam'
+        prod1 = self.env['product.product'].create({
+            'name': 'elct1',
+            'type': 'consu',
+            'route_ids': [(6, 0, [
+                warehouse.manufacture_pull_id.route_id.id,
+                warehouse.mto_pull_id.route_id.id
+            ])],
+        })
+        prod2 = self.env['product.product'].create({
+            'name': 'elct2',
+            'type': 'consu',
+            'route_ids': [(6, 0, [
+                warehouse.manufacture_pull_id.route_id.id,
+                warehouse.mto_pull_id.route_id.id
+            ])],
+        })
+        partner = self.env['res.partner'].create({'name': 'Steve Buscemi'})
+        so = self.env['sale.order'].create({
```

Comment:
You create a sale order in `mrp`. But this module does not depend on sale. Please move it into `sale_mrp` module